### PR TITLE
[frontendActions][reset] Precede action with info dialog (closes #442)

### LINKED ISSUE #442: Consider an alert with checkbox for less obvious actions (incl. reset to remote branch)
Like in Ctrl+T dialog:

image

### PR DESCRIPTION
**Pull request by mkondratek**
The info dialog:

image

### CODE REVIEW

File: frontendActions/src/main/resources/GitMacheteBundle.properties [hunk 1]
```diff
@@ -53,6 +53,11 @@ action.GitMachete.BaseRebaseBranchOntoParentAction.text=_Rebase Branch onto Pare
 action.GitMachete.BaseResetBranchToRemoteAction.description.enabled={0} branch ''{1}'' to its remote tracking branch
 action.GitMachete.BaseResetBranchToRemoteAction.action-name=Re_set to Remote
 action.GitMachete.BaseResetBranchToRemoteAction.description-action-name=Reset to remote
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.dont-show-again=Don't Show Again
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.title=Git Machete Reset Info
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.ok-text=Reset
+# TODO (#444): prohibit reset on conflicts with uncommitted
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.message=<html>Git Machete Reset performs <b>git reset --keep</b>.<br>In case of uncommitted changes you will be asked to resolve File Cache Conflicts.<br>Locally committed changes diverged from the remote will be lost.</html>
```
Comment (PawelLipski) [inline frontendActions/src/main/resources/GitMacheteBundle.properties:60]: Hmm I don't really like the `...will be lost` phrasing :/

Comment (PawelLipski) [reply frontendActions/src/main/resources/GitMacheteBundle.properties:60]: Maybe sth along the lines `<branch-name> will point to the same commit as its remote branch. Use `git reflog <branch-name> to see the history of commits pointed by '<branch-name>'`... a bit awkard, I guess a better formulation is possible

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java [hunk 2]
```diff
@@ -35,6 +39,8 @@
       IExpectsKeyProject,
       ISyncToRemoteStatusDependentAction {
 
+  public static final String RESET_INFO_SHOWN = "git-machete.reset.info.shown";
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:42]: Hmm is it the convention for `PropertiesComponent` keys? For instance the VCS related ones have UPPER_CASE:

```js
    propertiesComponent.setValue('ASKED_ADD_EXTERNAL_FILES', true);
    propertiesComponent.setValue('ASKED_SHARE_PROJECT_CONFIGURATION_FILES', true);
```

Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:42]: there is no convention XD 
```
REQUEST_FEEDBACK = "request-lint-feedback";
// and in the other place
String PREFERENCE_KEY_PREFIX = "ScenePreference";
String SHOW_TOOLTIP_KEY = PREFERENCE_KEY_PREFIX + "ShowToolTip";
```

But we can use the VCS version as it is closer to us

Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:42]: Heh okay ;D use your judgement

**Comment by mkondratek** [comment]
Current version:
image

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java (hunk 2 above)
Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:42]: It is not consistent even within VCS... :c leaving as it is

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java [hunk 3]
```diff
@@ -112,6 +118,20 @@ public void actionPerformed(AnActionEvent anActionEvent) {
       return;
     }
 
+    if (!PropertiesComponent.getInstance().getBoolean(RESET_INFO_SHOWN)) {
+      final var showOkCancelDialog = MessageUtil.showOkCancelDialog(
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:122]:
```suggestion
      final var okCancelDialogResult = MessageUtil.showOkCancelDialog(
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java [hunk 4]
```diff
@@ -112,6 +118,20 @@ public void actionPerformed(AnActionEvent anActionEvent) {
...
+          GitMacheteBundle.getString("action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.title"),
+          GitMacheteBundle.message("action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.message", branchName.get()),
+          GitMacheteBundle.getString("action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.ok-text"),
+          Messages.getCancelButton(),
+          Messages.getWarningIcon(),
+          new ResetBranchToRemoteInfoDialog(),
+          project);
+      if (showOkCancelDialog != 0) {
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:130]:
```suggestion
      if (okCancelDialogResult != Messages.OK) {
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java [hunk 5]
```diff
@@ -116,6 +122,20 @@ public void actionPerformed(AnActionEvent anActionEvent) {
       return;
     }
 
+    if (!PropertiesComponent.getInstance().getBoolean(RESET_INFO_SHOWN)) {
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetBranchToRemoteAction.java:125]: `getBoolean` returns `false` (and doesn't e.g. throw an exception) if a key is missing? Maybe add a comment that states that explicitly

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties [hunk 6]
```diff
@@ -57,6 +57,17 @@ action.GitMachete.BaseRebaseBranchOntoParentAction.text=_Rebase Branch onto Pare
 action.GitMachete.BaseResetBranchToRemoteAction.description.enabled={0} branch ''{1}'' to its remote tracking branch
 action.GitMachete.BaseResetBranchToRemoteAction.action-name=Re_set to Remote
 action.GitMachete.BaseResetBranchToRemoteAction.description-action-name=Reset to remote
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.dont-show-again=Don't Show Again
```
Comment (PawelLipski) [inline frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:60]: Hmm why single single quote here but double single quote in other cases? :thinking:

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties [hunk 7]
```diff
@@ -57,6 +57,17 @@ action.GitMachete.BaseRebaseBranchOntoParentAction.text=_Rebase Branch onto Pare
...
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.title=Git Machete Reset Info
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.ok-text=_Reset
+# TODO (#444): prohibit reset on conflicts with uncommitted
+action.GitMachete.BaseResetBranchToRemoteAction.info-dialog.message=\
+  <html>\
```
Comment (PawelLipski) [inline frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:65]: `<html>` is necessary here but not in the next one (`notification.success`)? :thinking:

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties (hunk 6 above)
Comment (mkondratek) [reply frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:60]: The pair of single-quotes escape format placeholders (like `{0}`). Additionally, double single quote reduces to a single single-quote. Single signle-quote is just a single single-quote. :rofl: 

Eventually, it looks correctly (see the image above).

Comment (PawelLipski) [reply frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:60]: Aaaah okay, so the only potential error would be to first declare prop as `hello '{0}'` and then try to pass argument to this escaped '{0}', right? This should be anyway captured by I18nFormatChecker (you can verify at your convenience btw)

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties (hunk 7 above)
Comment (mkondratek) [reply frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:65]: ¯\_(ツ)_/¯

So there are action descriptions (probably the same for action texts) - no formatting, that is why we use single quotes for branch names there.
There are notification messages - we can use <b></b> tags.
There are dialogs messages (aka content) - html wrapping is needed to "display" tags properly.

Comment (PawelLipski) [reply frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:65]: Whooops... pls drop a comment about this oddities then somewhere in the file

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties (hunk 6 above)
Comment (mkondratek) [reply frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:60]: verified

**Pull request merged**